src/backend/glutin.rs

```diff
@@ -143,44 +147,126 @@ pub enum GlutinBackendError
     WindowClosed
 }
 
-pub type GlutinInputId = ();
-pub type GlutinInputBackendHandler<T: InputBackendHandler<GlutinInputId>> = T;
-
 pub struct GlutinInputBackend
 {
     window: Rc<Window>,
-    pub handler: Option<GlutinInputBackendHandler>,
+    time_counter: u32,
+    handler: Option<Box<InputBackendHandler>>,
 }
 
 impl GlutinInputBackend
 {
-    pub fn process_new_events(&self) -> Result<(), GlutinBackendError>
+    pub fn set_handler<I: InputBackendHandler + 'static>(&mut self, handler: I) {
+        self.handler = Some(Box::new(handler));
+    }
+
+    pub fn clear_handler(&mut self) {
+        self.handler = None;
+    }
+
+    pub fn process_new_events(&mut self) -> Result<(), GlutinBackendError>
     {
         for event in self.window.poll_events()
         {
-            if let Some(handler) = self.handler {
+            if let Some(ref mut handler) = self.handler {
                 match event {
-                    Event::MouseEntered => {
-                        handler.on_input_attached(InputDevice::Mouse(()));
-                    },
-                    Event::MouseLeft => {
-                        handler.on_input_detached(InputDevice::Mouse(()));
-                    },
                     Event::Focused(focused) => if focused {
-                        handler.on_input_attached(InputDevice::Keyboard(()));
+                        handler.on_seat_created(&0u32);
```

Comment:
This was done mostly because I was too lazy to do this outside of `process_events` and because it helps testing, if you don't need to restart to trigger this event. In theory a compositor should be able to handle this, but I see your point. I will change it.